core/plugins/org.fusesource.ide.foundation.ui/src/org/fusesource/ide/foundation/ui/util/ControlDecorationHelper.java

```diff
@@ -18,12 +18,34 @@
 
 public class ControlDecorationHelper {
 
-	public void addInformationOnFocus(Control control, String informationText) {
-		ControlDecoration decoration = new ControlDecoration(control, SWT.BOTTOM | SWT.LEFT);
-		Image image = FieldDecorationRegistry.getDefault().getFieldDecoration(FieldDecorationRegistry.DEC_INFORMATION).getImage();
-		decoration.setImage(image);
-		decoration.setShowOnlyOnFocus(true);
-		decoration.setDescriptionText(informationText);
+	public ControlDecoration addInformationOnFocus(Control control, String informationText) {
+		ControlDecoration decoration = null;
+		if (control != null) {
+			decoration = new ControlDecoration(control, SWT.BOTTOM | SWT.LEFT);
+			Image image = FieldDecorationRegistry.getDefault().getFieldDecoration(FieldDecorationRegistry.DEC_INFORMATION).getImage();
+			decoration.setImage(image);
+			decoration.setShowOnlyOnFocus(true);
+			decoration.setDescriptionText(informationText);
+		}
+		return decoration;
 	}
 	
+	public ControlDecoration addErrorToControl(Control control, String informationText) {
```

Comment:
parameter might be better named with something like errorMessage or validationMessage

Reply:
agreed